src/servers/grpc_server_v2.cc

```diff
@@ -1315,24 +1362,163 @@ InferGRPCToInput(
                 .c_str());
       }
 
-      // FIXMEV2 handle non-raw content types
+      // Try to read the raw contents if available
       const std::string& raw = io.contents().raw_contents();
       base = raw.c_str();
       byte_size = raw.size();
+
+      // Check the presence of explicit tensors
+      if (io.contents().bool_contents_size() != 0) {
+        RETURN_IF_ERR(InferGRPCToInputHelper(
+            io.name(), request.model_name(), "BOOL", io.datatype(), byte_size));
+        base = (const void*)io.contents().bool_contents().data();
+        byte_size = io.contents().bool_contents_size();
+      }
+
+      if (io.contents().int_contents_size() != 0) {
+        if (io.datatype().compare("INT8") == 0) {
+          RETURN_IF_ERR(InferGRPCToInputHelper(
+              io.name(), request.model_name(), "INT8", io.datatype(),
+              byte_size));
+          std::shared_ptr<std::string> serialized(new std::string());
```

Comment:
You should use string::reserve() to size the string to what is expected for the string so that it doesn't need to realloc.

Reply:
Done